controllers/helmrepository_controller_oci.go

```diff
@@ -292,20 +291,25 @@ func (r *HelmRepositoryOCIReconciler) reconcileSource(ctx context.Context, obj *
 		}
 	}
 
-	if result, err := r.validateSource(ctx, obj, loginOpts...); err != nil || result == sreconcile.ResultEmpty {
-		return result, err
-	}
-
-	return sreconcile.ResultSuccess, nil
+	return r.validateSource(ctx, obj, loginOpts...)
 }
 
 // validateSource the HelmRepository object by checking the url and connecting to the underlying registry
 // with he provided credentials.
 func (r *HelmRepositoryOCIReconciler) validateSource(ctx context.Context, obj *sourcev1.HelmRepository, logOpts ...helmreg.LoginOption) (sreconcile.Result, error) {
+	if !helmreg.IsOCI(obj.Spec.URL) {
+		e := &serror.Stalling{
+			Err:    fmt.Errorf("the url scheme is not supported: %s", obj.Spec.URL),
+			Reason: sourcev1.URLInvalidReason,
+		}
+		conditions.MarkFalse(obj, meta.ReadyCondition, e.Reason, e.Err.Error())
+		return sreconcile.ResultEmpty, e
+	}
```

Comment:
Looking at the overall behavior, it may be better to perform this check at the very beginning in `reconcileSource()` before doing anything related to secret fetching and authentication as all that would be of no use if the URL isn't valid.